pkg/autostart/managers.go

```diff
@@ -0,0 +1,167 @@
+// SPDX-FileCopyrightText: Copyright The Lima Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package autostart manage start at login unit files for darwin/linux
+package autostart
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/lima-vm/lima/v2/pkg/autostart/launchd"
+	"github.com/lima-vm/lima/v2/pkg/autostart/systemd"
+	"github.com/lima-vm/lima/v2/pkg/limatype"
+	"github.com/lima-vm/lima/v2/pkg/textutil"
+)
+
+type notSupportedManager struct{}
+
+var ErrNotSupported = fmt.Errorf("autostart is not supported on %s", runtime.GOOS)
+
+func (*notSupportedManager) IsRegistered(_ context.Context, _ *limatype.Instance) (bool, error) {
+	return false, ErrNotSupported
+}
+
+func (*notSupportedManager) RegisterToStartAtLogin(_ context.Context, _ *limatype.Instance) error {
+	return ErrNotSupported
+}
+
+func (*notSupportedManager) UnregisterFromStartAtLogin(_ context.Context, _ *limatype.Instance) error {
+	return ErrNotSupported
+}
+
+func (*notSupportedManager) AutoStartedIdentifier() string {
+	return ""
+}
+
+func (*notSupportedManager) RequestStart(_ context.Context, _ *limatype.Instance) error {
+	return ErrNotSupported
+}
+
+func (*notSupportedManager) RequestStop(_ context.Context, _ *limatype.Instance) (bool, error) {
+	return false, ErrNotSupported
+}
+
+// Launchd is the autostart manager for macOS.
+var Launchd = &TemplateFileBasedManager{
+	filePath:              launchd.GetPlistPath,
+	template:              launchd.Template,
+	enabler:               launchd.EnableDisableService,
+	autoStartedIdentifier: launchd.AutoStartedServiceName,
+	requestStart:          launchd.RequestStart,
+	requestStop:           launchd.RequestStop,
+}
+
+// Systemd is the autostart manager for Linux.
+var Systemd = &TemplateFileBasedManager{
+	filePath:              systemd.GetUnitPath,
+	template:              systemd.Template,
+	enabler:               systemd.EnableDisableUnit,
+	autoStartedIdentifier: systemd.AutoStartedUnitName,
+	requestStart:          systemd.RequestStart,
+	requestStop:           systemd.RequestStop,
+}
```

Comment:
This feels weird. Why is `TemplateFileBasedManager` not a regular Go interface?

Reply:
`TemplateFileBasedManager` is one of the implementations of the `type autoStartManager interface`.
The differences between platforms are divided into `pkg/autostart/launchd` and `pkg/autostart/systemd`.
`TemplateFileBasedManager` has a common part that is not platform-dependent.